Application/EdFi.Ods.SwaggerUI/ConfigureMetadata.cs

```diff
@@ -0,0 +1,90 @@
+﻿// SPDX-License-Identifier: Apache-2.0
+// Licensed to the Ed-Fi Alliance under one or more agreements.
+// The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
+// See the LICENSE and NOTICES files in the project root for more information.
+
+using System;
+using System.Linq;
+using System.Net.Http;
+using System.Text.Json;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Swashbuckle.AspNetCore.SwaggerUI;
+
+namespace EdFi.Ods.SwaggerUI
+{
+    public class ConfigureMetadata : IConfigureOptions<SwaggerUIOptions>
+    {
+        private readonly ILogger _logger;
+
+        public ConfigureMetadata(ILogger<ConfigureMetadata> logger)
+        {
+            _logger = logger;
+        }
+
+        public void Configure(SwaggerUIOptions options)
+        {
+            AddMetadataEndpoints(options);
+        }
+
+        private void AddMetadataEndpoints(SwaggerUIOptions options)
+        {
+            if (!options.ConfigObject.AdditionalItems.TryGetValue("WebApiVersionUrl", out object webApiVersionUrl))
+            {
+                _logger.LogError($"'WebApiVersionUrl' property in appsettings.json was not found, skipping metadata retrieval.");
+                return;
+            }
+
+            try
+            {
+                using var httpClient = new HttpClient();
+
+                var versionResponse = httpClient
```

Comment:
my concern is this, why not do it as an async task and then return the metadata json before the final enumeration